### PR TITLE
⚠️ E2E framework resolves CNI_RESOURCES without using env vars

### LINKED ISSUE #3874: Forward-port E2E test now resolve CNI_RESOURCES without using env variables
https://github.com/kubernetes-sigs/cluster-api/pull/3846 improved E2E test to resolve CNI_RESOURCES without using env variables in the v0.3 branch;
These changes should now be forward ported to v0.4

/kind feature

### PR DESCRIPTION
**Pull request by arghya88**
**What this PR does / why we need it**:
Forward Ports #3846 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3874

### CODE REVIEW

**Comment by arghya88** [comment]
APIDiff

```
sigs.k8s.io/cluster-api/test/framework/clusterctl
  Incompatible changes:
  - CreateRepositoryInput: old is comparable, new is not
  Compatible changes:
  - (*CreateRepositoryInput).RegisterClusterResourceSetConfigMapTransformation: added
  - CreateRepositoryInput.FileTransformations: added
  - RepositoryFileTransformation: added
+ EXIT_VALUE=1
+ set +o xtrace
```

File: test/framework/clusterctl/e2e_config.go [hunk 1]
```diff
@@ -66,15 +65,11 @@ func LoadE2EConfig(ctx context.Context, input LoadE2EConfigInput) *E2EConfig {
 
 // SetCNIEnvVar read CNI from cniManifestPath and sets an environmental variable that keeps CNI resources.
 // A ClusterResourceSet can be used to apply CNI using this environmental variable.
+//
+// Deprecated: Use FileTransformations in the CreateRepositoryInput to embedded CNI into cluster templates during create repository.
```
Comment (fabriziopandini) [inline test/framework/clusterctl/e2e_config.go:69]: Being this func deprecated in v0.3.x, it is now possible to remove it in the v0.4.0 branch

**Review by fabriziopandini — COMMENTED** [review]
lgtm pending addressing of the two comments

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**